rosidl_generator_c/test/test_interfaces.c

```diff
@@ -0,0 +1,821 @@
+
+#include "rosidl_generator_c/msg/various.h"
+#include "rosidl_generator_c/msg/bool.h"
+#include "rosidl_generator_c/msg/byte.h"
+#include "rosidl_generator_c/msg/char.h"
+#include "rosidl_generator_c/msg/float32.h"
+#include "rosidl_generator_c/msg/float64.h"
+#include "rosidl_generator_c/msg/int8.h"
+#include "rosidl_generator_c/msg/int16.h"
+#include "rosidl_generator_c/msg/int32.h"
+#include "rosidl_generator_c/msg/int64.h"
+#include "rosidl_generator_c/msg/uint8.h"
+#include "rosidl_generator_c/msg/uint16.h"
+#include "rosidl_generator_c/msg/uint32.h"
+#include "rosidl_generator_c/msg/uint64.h"
+#include "rosidl_generator_c/msg/primitive_values.h"
+#include "rosidl_generator_c/msg/strings.h"
+#include "rosidl_generator_c/msg/primitives_unbounded_arrays.h"
+#include "rosidl_generator_c/msg/primitives_bounded_arrays.h"
+#include "rosidl_generator_c/msg/primitives_static_arrays.h"
+#include "rosidl_generator_c/msg/telegram1.h"
+#include "rosidl_generator_c/msg/telegram2.h"
+#include "rosidl_generator_c/msg/wire.h"
+
+#define TEST_STRING \
+  "Deep into that darkness peering, long I stood there wondering, fearing"
+
+#define ARRAY_SIZE 7
+
+void test_primitives(void);
+void test_primitives_default_value(void);
+void test_strings(void);
+void test_primitives_unbounded_arrays(void);
+void test_primitives_bounded_arrays(void);
+void test_primitives_static_arrays(void);
+void test_submessages(void);
+
+int main(void)
+{
+  fprintf(stderr, "Testing rosidl_generator_c message types...\n");
+  fprintf(stderr, "Testing simple primitive message types...\n");
+  test_primitives();
+  fprintf(stderr, "Testing simple primitives with default values...\n");
+  test_primitives_default_value();
+  fprintf(stderr, "Testing string types...\n");
+  test_strings();
+  fprintf(stderr, "Testing primitives unbounded arrays types...\n");
+  test_primitives_unbounded_arrays();
+  fprintf(stderr, "Testing primitives bounded arrays types...\n");
+  test_primitives_bounded_arrays();
+  fprintf(stderr, "Testing primitives static arrays types...\n");
+  test_primitives_static_arrays();
+  fprintf(stderr, "Testing nested sub-messages...\n");
+  test_submessages();
+  fprintf(stderr, "All tests were good!\n");
+  return 0;
+}
+
+/**
+ * Test message with simple primitive types.
+ */
+void test_primitives(void)
+{
+  rosidl_generator_c__msg__Bool bool_msg;
+  rosidl_generator_c__msg__Byte byte_msg;
+  rosidl_generator_c__msg__Char char_msg;
+  rosidl_generator_c__msg__Float32 float32_msg;
+  rosidl_generator_c__msg__Float64 float64_msg;
+  rosidl_generator_c__msg__Int8 int8_msg;
+  rosidl_generator_c__msg__Int16 int16_msg;
+  rosidl_generator_c__msg__Int32 int32_msg;
+  rosidl_generator_c__msg__Int64 int64_msg;
+  rosidl_generator_c__msg__Uint8 uint8_msg;
+  rosidl_generator_c__msg__Uint16 uint16_msg;
+  rosidl_generator_c__msg__Uint32 uint32_msg;
+  rosidl_generator_c__msg__Uint64 uint64_msg;
+
+  bool_msg.empty_bool = true;
+  assert(bool_msg.empty_bool == true);
```

Comment:
`assert` is not a viable way to test in a unit test. When build in release mode none of these checks is performed.